### DISCUSSION TITLE
Causal MMM and do() operator

### DISCUSSION THREAD

**Discussion by AlfredoJF** [pymc-labs/pymc-marketing#952] [Q&A]
Does anyone have an example of a causal model using the `MMM` class? Or this should be done as in the components notebook?

I had a look at this blog post https://www.pymc-labs.com/blog-posts/causal-analysis-with-pymc-answering-what-if-with-the-new-do-operator/ and was wondering if we could create a causal model and test some "What-if" scenarios to test causality.

I know adding lift studies is available but I would like to test this in the meantime the experiments are done.

**Comment by cluhmann** [comment]
Depends what exactly you are looking for. There are a couple of examples [here](https://www.pymc-marketing.io/en/stable/notebooks/mmm/mmm_tvp_example.html#simulate-sales) and [here](https://www.pymc-marketing.io/en/stable/notebooks/mmm/mmm_time_varying_media_example.html#temporal-hidden-latent-process).

**Reply by AlfredoJF** [reply]
Thanks @cluhmann that helps with the do() operator question. What about causal models built on top of pymc-marketing?

And by causal I meant confounders that influence media and sales.

**Reply by cluhmann** [reply]
Again, it sort of depends on what you are envisioning and what you want to accomplish. But I might imagine that you are actually talking about building "underneath" your MMM. That is, confounders could be modeled as latent (i.e., unmeasured) factors reflected in two or more observed variables. Then one could add these latent factors as drivers. That would be pretty involved and would probably require diving into some PyMC models, but others here may have some more straightforward suggestions.
